flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/sink/global/LocalCommitterOperator.java

```diff
@@ -0,0 +1,205 @@
+import org.apache.flink.runtime.state.StateInitializationContext;
+import org.apache.flink.streaming.runtime.operators.sink.committables.CommitRequestImpl;
+import org.apache.flink.streaming.runtime.operators.sink.committables.CommitRequestState;
+import org.apache.flink.util.function.SerializableSupplier;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/** A {@link AbstractCommitterOperator} to process local committer. */
+public class LocalCommitterOperator<CommT> extends AbstractCommitterOperator<CommT, CommT> {
+
+    private static final long serialVersionUID = 1L;
+
+    private final SerializableSupplier<Committer<CommT>> committerFactory;
+
+    private Committer<CommT> committer;
+
+    public LocalCommitterOperator(
+            SerializableSupplier<Committer<CommT>> committerFactory,
+            SerializableSupplier<SimpleVersionedSerializer<CommT>> committableSerializer) {
+        super(committableSerializer);
+        this.committerFactory = checkNotNull(committerFactory);
+    }
+
+    @Override
+    public void initializeState(StateInitializationContext context) throws Exception {
+        committer = committerFactory.get();
+        super.initializeState(context);
+    }
+
+    @Override
+    public void commit(boolean isRecover, List<CommT> committables)
+            throws IOException, InterruptedException {
+        if (committables.isEmpty()) {
+            return;
+        }
+
+        List<CommitRequestImpl> requests = new ArrayList<>(committables.size());
+        for (CommT comm : committables) {
+            requests.add(new CommitRequestImpl(comm));
+        }
+
+        long sleep = 1000;
+        while (true) {
+            // commit
+            requests.forEach(CommitRequestImpl::setSelected);
+            committer.commit(new ArrayList<>(requests));
+            requests.forEach(CommitRequestImpl::setCommittedIfNoError);
+
+            // drain finished
+            requests.removeIf(CommitRequestImpl::isFinished);
+            if (requests.isEmpty()) {
+                return;
+            }
+
+            //noinspection BusyWait
+            Thread.sleep(sleep);
+            sleep *= 2;
+        }
+    }
+
+    @Override
+    public List<CommT> toCommittables(long checkpoint, List<CommT> inputs) {
+        return inputs;
+    }
+
+    @Override
+    public void close() throws Exception {
+        committer.close();
+        super.close();
+    }
+
+    /** {@link CommitRequest} implementation. */
+    public class CommitRequestImpl implements CommitRequest<CommT> {
+
+        private CommT committable;
+        private int numRetries;
+        private CommitRequestState state;
+
+        private CommitRequestImpl(CommT committable) {
+            this.committable = committable;
+            this.state = CommitRequestState.RECEIVED;
```

Comment:
Nit: since each request goes through `requests.forEach(CommitRequestImpl::setSelected)`, there's no need to init `this.state` to `CommitRequestState.RECEIVED`

Reply:
Anyway, It needs to be initialized, and we don't want to set a null value